src/Sarif.UnitTests/Writers/SarifLoggerTests.cs

```diff
@@ -420,5 +404,161 @@ public void SarifLogger_ResultAndRuleIdMismatch()
                 sarifLogger.Log(rule, result);
             }
         }
+
+        [TestMethod]
+        public void SarifLogger_LoggingOptions_ComputeFileHashes()
+        {
+            TestForLoggingOption(LoggingOptions.ComputeFileHashes);
+        }
+
+
+        [TestMethod]
+        public void SarifLogger_LoggingOptions_None()
+        {
+            TestForLoggingOption(LoggingOptions.None);
+        }
+
+        [TestMethod]
+        public void SarifLogger_LoggingOptions_PersistEnvironment()
+        {
+            TestForLoggingOption(LoggingOptions.PersistEnvironment);
+        }
+
+        [TestMethod]
+        public void SarifLogger_LoggingOptions_PersistFileContents()
+        {
+            TestForLoggingOption(LoggingOptions.PersistFileContents);
+        }
+
+        [TestMethod]
+        public void SarifLogger_LoggingOptions_PrettyPrint()
+        {
+            TestForLoggingOption(LoggingOptions.PrettyPrint);
+        }
+
+        [TestMethod]
+        public void SarifLogger_LoggingOptions_Verbose()
+        {
+            TestForLoggingOption(LoggingOptions.Verbose);
+        }
+
+        [TestMethod]
+        public void SarifLogger_LoggingOptions_All()
+        {
+            TestForLoggingOption(LoggingOptions.All);
+        }
+
+        [TestMethod]
+        public void SarifLogger_LoggingOptions_Count()
+        {
+            // This test exists in order to alert test developers when a new member is added to the
+            // LoggingOptions enum. In that case, this test and others should be updated/added
+            // to account for the new member.
+            //     
+            // Current values are:
+            // None, ComputeFileHashes, PersistEnvironment, PersistFileContents, PrettyPrint, Verbose, All  
+            Enum.GetNames(typeof(LoggingOptions)).Length.Should().Be(7);
+        }
+
+        private void TestForLoggingOption(LoggingOptions loggingOption)
+        {
+            string fileName = Path.GetTempFileName();
+
+            try
+            {
+                SarifLogger logger;
+
+                using (logger = new SarifLogger(fileName, loggingOption))
+                {
+                    ValidateLoggerForExclusiveOption(logger, loggingOption);
+                };
+
+                var sb = new StringBuilder();
+                var stringWriter = new StringWriter(sb);
+                using (logger = new SarifLogger(stringWriter, loggingOption))
+                {
+                    ValidateLoggerForExclusiveOption(logger, loggingOption);
+                };
+            }
+            
+            finally
+            {
+                if (File.Exists(fileName)) { File.Delete(fileName); }
```

Comment:
> { [](start = 42, length = 2)

convention: braces on separate lines

Reply:
I see you're using this convention alot in this PR. I've never seen you use it before.

---
In reply to: [135838612](https://github.com/Microsoft/sarif-sdk/pull/609#discussion_r135838612) [](ancestors = 135838612)

Reply:
If a line can be clearly expressed on a single line, I use this convention (which also must included braces). you can 'chunk' this line easily, and it minimizes vertical whitespace

Reply:
cited in our code standard doc. single line statements ok as long as it is readable and all possible braces are expressed.

---
In reply to: [135841637](https://github.com/Microsoft/sarif-sdk/pull/609#discussion_r135841637) [](ancestors = 135841637,135838612)